scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_table_widget_view.py

```diff
@@ -0,0 +1,328 @@
+
+        self.grouping_table.horizontalHeaderItem(4).setToolTip("A list of Rebins :"
+                                                               "\n  - Select None, Fixed or Variable from the list.")
+        self.grouping_table.horizontalHeaderItem(5).setToolTip("Rebin Options :"
+                                                               "\n  For fixed rebin enter number of steps"
+                                                               "\n  For variable rebin enter Bin Boundaries")
+
+    def num_rows(self):
+        return self.grouping_table.rowCount()
+
+    def num_cols(self):
+        return self.grouping_table.columnCount()
+
+    def notify_data_changed(self):
+        if not self._updating:
+            self.dataChanged.emit()
+
+    def get_index_of_text(self, selector, text):
+        for i in range(selector.count()):
+            if str(selector.itemText(i)) == text:
+                return i
+        return 0
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Adding / removing table entries
+    # ------------------------------------------------------------------------------------------------------------------
+    def add_entry_to_table(self, row_entries):
+        assert len(row_entries) == self.grouping_table.columnCount()
+        row_position = self.grouping_table.rowCount()
+        self.grouping_table.insertRow(row_position)
+        for i, entry in enumerate(row_entries):
+            table_item = QtWidgets.QTableWidgetItem(entry)
+            table_item.setFlags(table_column_flags[group_table_columns[i]])
+
+            if group_table_columns[i] == 'to_analyse':
+                if entry:
+                    table_item.setCheckState(QtCore.Qt.Checked)
+                else:
+                    table_item.setCheckState(QtCore.Qt.Unchecked)
+
+            if group_table_columns[i] == 'rebin':
+                rebin_selector_widget = self._rebin_selection_cell_widget()
+                rebin_selector_widget.currentIndexChanged.connect(
+                     lambda i, row_position=row_position: self.on_rebin_combo_changed(i, row_position))
+                index = int(table_item.text())
+                rebin_selector_widget.setCurrentIndex(index)
+                self.grouping_table.setCellWidget(row_position, i, rebin_selector_widget)
+
+            self.grouping_table.setItem(row_position, i, table_item)
+
+    def _get_selected_row_indices(self):
+        return list(set(index.row() for index in self.grouping_table.selectedIndexes()))
+
+    def get_selected_group_names(self):
+        indexes = self._get_selected_row_indices()
+        return [str(self.grouping_table.item(i, 0).text()) for i in indexes]
+
+    def remove_selected_groups(self):
+        indices = self._get_selected_row_indices()
+        for index in reversed(sorted(indices)):
```

Comment:
```suggestion
        for index in sorted(indices,reverse = True):
```
This can also be done like this